File: 백준/Silver/5567. 결혼식/결혼식.js
function dfs(friends, visited, cur, depth) {
  if (depth === 2) {
    return
  }

  for (let i of friends[cur]) {
    if (!visited[i]) {
      visited[i] = true
    }
    dfs(friends, visited, i, depth+1)
  }
}

const fs = require('fs')
// const input = fs.readFileSync('input.txt').toString().trim().split('\n')
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')

const n = parseInt(input[0])
const m = parseInt(input[1])

let visited = new Array(n+1).fill(false)
visited[1] = true

let friends = Array.from({ length: n + 1 }, () => []);
for (let i = 2; i < 2+m; i++) {
  let [a, b] = input[i].trim().split(' ').map(Number)
  if (!friends[a]) friends[a] = [];
  if (!friends[b]) friends[b] = [];
  friends[a].push(b)
  friends[b].push(a)
}

dfs(friends, visited, 1, 0)
// console.log(friends)
// console.log(visited)
console.log(visited.filter(x=>x).length-1)

File: 백준/Silver/5567. 결혼식/결혼식.py
def dfs(frd, vis, cur, depth):
    if depth == 2:
        return
    
    for i in frd[cur]:
        if not vis[i]:
            vis[i] = True
        dfs(frd, vis, i, depth+1)

n = int(input())
m = int(input())
friends = [map(int, input().split()) for _ in range(m)]

frd = [[] for _ in range(n+1)]
vis = [False] * (n+1)
vis[1] = True

for a, b in friends:
    frd[a].append(b)
    frd[b].append(a)
    
dfs(frd, vis, 1, 0)
print(len([x for x in vis if x]) -1)